scripts/jenkins/cloud/ansible/roles/setup_zypper_repos/defaults/main.yml

```diff
@@ -15,7 +15,27 @@
 #
 ---
 
-setup_distribution_minor_version: "{{ ansible_distribution_version.split('.')[1] }}"
+#
+# zypper repo management
+#
+zypper_repo_enabled: yes
+zypper_repo_state: "present"
+# set to true to delete repo backing content from the system;
+# only valid if enabled==no and state==absent
+zypper_repo_delete: false
+
+setup_cloud_info:
+  cloud9:
+    sles_version: 12.4
+  cloud8:
+    sles_version: 12.3
+  cloud7:
+    sles_version: 12.2
+
+setup_sles_version: "{{ setup_cloud_info[cloud_release].sles_version }}"
+setup_sles_major_version: "{{ setup_sles_version.split('.')[0] }}"
+setup_sles_minor_version: "{{ setup_sles_version.split('.')[1] }}"
+setup_suse_release: "suse-{{ setup_sles_version }}"
```

Comment:
Just a question. Why do you need to add a `setup_sles_version` instead of using what is in group_vars/all/all.yml?
https://github.com/rtamalin/automation/blob/zypper_repo_disable_or_remove/scripts/jenkins/cloud/ansible/group_vars/all/all.yml#L62

Reply:
So there are a few reasons why I settled on this approach for handling managing the version determination, not all of which are apparent in this patch, but which are related to how I got things to work for the ardana upgrade in PR#3732 which is based upon this patch set; I broke down the huge set of upgrade changes into a sequence of 4 or 5 patches some of which were independent of each other.

Basically the way we manage specifying the job parameters causes problems for trying to dynamically override settings in the ansible code; because we use `--extra-args @input.yml` on the command line, any value present in the input.yml supercede occurences in group_vars, host_vars, roles/<role>/vars and roles/<role>/defaults.

Additionally we also run into issues with ansible fact gathering persistence settings; they persist for 24hours or more which makes it hard to rely on the ansible gathered facts when we want to upgrade from SP3/Cloud8 to SP4/Cloud9... And then there is the fact that one you add the SP4 Pool repo to a system, if you sample the /etc/os-release file it shows the SP4 settings, even if you haven't upgraded the system yet...

And yes we can move some of the settings used here into the global group_vars/all space, but the pattern that prevails in other roles is that "role" specific globals tend to live in the roles defaults area rather than the group_vars/all area...

Note that once I get the ardana upgrade workflow landed I plan some time to address the issues with how we use input.yml, e.g. move the relevant vars to an input.yml under the group_vars/all space, and then test replacing that input.yml with the job specific settings by overwriting that file, rather than using the --extra-args approach...

Reply:
ok thx. I m fine with that.